src/hooks/use-accordion.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react'
+
+type AccordionState = number[] | number | null
+type SetAccordionState = (value: number) => void
+
+interface Options {
+  toggle?: boolean
+}
+
+function useAccordions(
+  initialState: number[],
+  options?: Options
+): [number[], SetAccordionState]
+
+function useAccordions(
+  initialState: number,
+  options: { toggle: false }
+): [number, SetAccordionState]
+
+function useAccordions(
+  initialState?: number | null,
+  options?: Options
+): [number | null, SetAccordionState]
```

Comment:
You can also consider an approach as in reducers for example, by specifying a type of accordion property. Then we can infer correct type of value, but current approach is also pretty solid for solving this problem. What do you think?

Reply:
Thanks for suggestion! I was really unsure how to properly type this hook. I made some changes and added multiple property because it is much more intuitive for other developers than previous variant. I guess it would be better to avoid function overloading and use generics with infer instead. But I still couldn't make types work properly with generics and infer 😓 (can be solved with `as` but we don't use it in our project). Maybe, we can discuss it during the meeting.

Reply:
I see. Using `generics` is definitely a better option to use. As far as I know, using `any` or `as` casting is permissible inside custom hooks or utility functions, especially if they provide different signatures. The main thing is to output correct type and the typescript internals do not matter

Reply:
I replaced overloading with generics, it looks better now.